mlflow/models/utils.py

```diff
@@ -938,6 +960,13 @@ def _is_scalar(x):
             if extra_cols:
                 message += f" Note that there were extra inputs: {extra_cols}"
             raise MlflowException(message)
+        # check for null values in required columns of Pyspark DataFrame
+        if isinstance(original_pf_input, SparkDataFrame):
+            for col in input_names:
+                if original_pf_input.filter(original_pf_input[col].isNull()).count() > 0:
```

Comment:
This could be a very expensive query depending on the size of the DataFrame. Do we strictly need to run this check?

Reply:
Recommended by @BenWilson2. Technically not necessary, but is useful for schema enforcement.